scipy/constants/_constants.py

```diff
@@ -329,10 +330,11 @@ def lambda2nu(lambda_: npt.ArrayLike) -> Any:
     array([  2.99792458e+08,   1.00000000e+00])
 
     """
-    return c / np.asanyarray(lambda_)
+    xp = array_namespace(lambda_)
+    return c / xp.asarray(lambda_)
```

Comment:
We have a float and an `xp` array. Should be either:

```suggestion
    return c / lambda_
```

or

```suggestion
    return xp.asarray(c) / lambda_
```

Reply:
I think these and the bellow might also been fine for the same reason as https://github.com/scipy/scipy/pull/20593#discussion_r1581759751

Reply:
Yes, as long as `a_float / an_xp_array == an_xp_array` in general.